conf/configuration.go

```diff
@@ -203,9 +229,48 @@ func MinPinners(ctx context.Context, db *database.DB) (int, error) {
 		return 0, err
 	}
 	if mp < minPinnersMinValue || mp > maxPinnersMinValue {
-		errMsg := fmt.Sprintf("Invalid min_pinners value in database configuration! The value must be between %d and %d, it was %v.", mp, minPinnersMinValue, maxPinnersMinValue)
+		errMsg := fmt.Sprintf("invalid min_pinners value in database configuration! The value must be between %d and %d, it was %v.", mp, minPinnersMinValue, maxPinnersMinValue)
 		build.Critical(errMsg)
 		return 0, errors.New(errMsg)
 	}
 	return int(mp), nil
 }
+
+// NextScan returns the time of the next cluster-wide scan for underpinned files.
+func NextScan(ctx context.Context, db *database.DB, logger logger.Logger) (time.Time, error) {
+	val, err := db.ConfigValue(ctx, ConfNextScan)
+	if errors.Contains(err, mongo.ErrNoDocuments) {
+		logger.Infof("Missing database value for '%s', setting a new one.", ConfNextScan)
+		// No scan has been scheduled. Schedule one in an hour.
+		err = SetNextScan(ctx, db, time.Now().Add(DefaultNextScanOffset).UTC())
+		if err != nil {
+			return time.Time{}, err
+		}
+		// Read the value from the DB in order to avoid race conditions.
+		// While this does not eliminate data races, it does help.
+		return NextScan(ctx, db, logger)
```

Comment:
The only way this is an endless loop is if MongoDB fails to insert a record in the DB *and* erroneously returns "no error". Similarly, the next `NextScan` recursive call can only be endless if MongoDB fails to write a valid RFC3339 time value somehow. Both situations are unlikely and depend of severe bugs in Mongo, which is not something we should reasonably code around.

In any case, this is moot because the recursive `NextScan` calls are removed.